### DISCUSSION TITLE
How to handle isolated protected routes with react navigation

### DISCUSSION THREAD

**Discussion by crisfcodes** [react-navigation/react-navigation#10557] [Q&A]
I am developing an app in which certain screens/features along the application need to be protected(accessed only for authenticated users).The react-navigation [docs](https://reactnavigation.org/docs/auth-flow/) shows an auth flow, this example is useful for those cases where it's mandatory to go through the login to be able to access all the application content. (render a set of screens if the user is logged in or not)

What I want is to protect specific screens where the user can run specific actions(like save a product) and only at that point redirect him to a login screen, if the user is successfully authenticated, redirect him to the point where he left off the flow. An important thing to highlight is, users can use the app and see some screens(product list/product detail) without being logged in. Here a simple image to illustrate the use case:

![Web 1280 – 1](https://user-images.githubusercontent.com/20000951/166100653-df9e48e4-20e4-49c7-9bc1-c5f945d57b03.png)

it is not clear to me how I can implement this authentication flow.

**Comment by luga97** [comment]
Hello, did you manage to solve this problem? I am also stuck with the same issue. I think creating an AuthNavigator inside the RootNavigator might solve this problem and allow us to handle navigation through route.params. However, I am not sure if this is the right approach or what happens with the auth navigation. Can users return to it? I hope we can get a response soon so I can determine if this is the correct solution

**Comment by ChamaraWP** [comment]
I facing same issue have you figure out how to do this
